aspnet/fundamentals/error-handling.rst

```diff
@@ -1,8 +1,171 @@
-.. include:: /../common/stub-topic.txt
+Error Handling
+==============
 
-|stub-icon| Error Handling
-==========================
+By `Steve Smith`_
 
-.. include:: /../common/stub-notice.txt
+When errors occur in your ASP.NET app, you can handle them in a variety of ways, as described in this article.
 
-.. _issue: https://github.com/aspnet/Docs/issues/124
\ No newline at end of file
+.. contents:: Sections
+	:local:
+	:depth: 1
+
+`View or download sample code <https://github.com/aspnet/Docs/tree/master/aspnet/fundamentals/error-handling/sample>`_
+
+Configuring an Exception Handling Page
+--------------------------------------
+
+You configure the pipeline for each request in the ``Startup`` class's ``Configure()`` method (learn more about :doc:`startup`). You can add a simple exception page, meant only for use during development, very easily. All that's required is to add a dependency on ``Microsoft.AspNet.Diagnostics`` to the project and then add one line to ``Configure()`` in ``Startup.cs``:
+
+.. literalinclude:: error-handling/sample/src/ErrorHandlingSample/Startup.cs
+	:language: c#
+	:lines: 21-29
+	:dedent: 8
+	:emphasize-lines: 6,8
+
+The above code includes a check to ensure the environment is development before adding the call to ``UseDeveloperExceptionPage``. This is a good practice, since you typically do not want to share detailed exception information about your application publicly while it is in production. :doc:`Learn more about configuring environments <environments>`.
+
+The sample application includes a simple mechanism for creating an exception:
+
+.. literalinclude:: error-handling/sample/src/ErrorHandlingSample/Startup.cs
+	:language: c#
+	:lines: 58-63
+	:dedent: 12
+	:emphasize-lines: 3-6
+
+If a request includes a non-empty querystring parameter for the variable ``throw`` (e.g. a path of ``/?throw=true``), an exception will be thrown. If the environment is set to ``Development``, the developer exception page is displayed:
+
+.. image:: error-handling/_static/developer-exception-page.png
+
+When not in development, it's a good idea to configure friendly pages for common HTTP status codes. You can use :ref:`status code pages <status-code-pages>` for this purpose.
```

Comment:
When not in development use the UseExceptionHandler* middleware instead.
https://github.com/aspnet/Templates/blob/dev/src/BaseTemplates/StarterWeb/Startup.cs#L51

The friendly status code pages are a separate feature, they can't deal with Exceptions.